homeassistant/components/aurora_abb_powerone/__init__.py

```diff
@@ -29,9 +32,30 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     comport = entry.data[CONF_PORT]
     address = entry.data[CONF_ADDRESS]
     serclient = AuroraSerialClient(address, comport, parity="N", timeout=1)
-
-    hass.data.setdefault(DOMAIN, {})[entry.unique_id] = serclient
-
+    # To handle yaml import attempts in darkeness, (re)try connecting only if
+    # unique_id not yet assigned.
+    if entry.unique_id is None:
+        try:
+            res = await hass.async_add_executor_job(
+                validate_and_connect, hass, entry.data
+            )
+        except AuroraError as error:
+            if "No response after" in str(error):
+                raise ConfigEntryNotReady("No response (could be dark)") from error
+        else:
+            # If we got here, the device is now communicating (maybe after
+            # being in darkness). But there's a small risk that the user has
+            # configured via the UI since we last attempted the yaml setup,
+            # which means we'd get a duplicate unique ID.
+            new_id = res[ATTR_SERIAL_NUMBER]
+            # Check if this unique_id has already been used
+            for existing_entry in hass.config_entries.async_entries(DOMAIN):
+                if existing_entry.unique_id == new_id:
+                    raise data_entry_flow.AbortFlow("already_configured")
```

Comment:
We can't raise AbortFlow here since we're not in the config flow. We should remove the current config entry with `hass.config_entries.async_remove` in a task, using `hass.async_create_task`, and return False.
```suggestion
                    _LOGGER.debug(
                        "Remove already configured config entry for id %s", new_id
                    )
                    hass.async_create_task(
                        hass.config_entries.async_remove(entry.entry_id)
                    )
                    return False
```

Reply:
Agreed. Done.